consensus/module.go

```diff
@@ -34,6 +35,11 @@ type ConsensusModule struct {
 	privateKey cryptoPocket.Ed25519PrivateKey
 	consCfg    modules.ConsensusConfig
 
+	// m is a mutex used to control synchronization when multiple goroutines are accessing the struct and its fields / properties.
```

Comment:
I think I figured out the answer to my own question based on your "smallest unit of work" idea.

Every method that implements a function from the consensus interface (i.e. `type ConsensusModule interface`) needs to acquire & defer a lock.

I think we should put this as a comment in `shared/modules/consensus_module.go`. Something like: "NOTE: Consensus is the core of the replicated state machine and is driven by various asynchronous events. Consider adding a mutex to your implementation that is acquired at the beginning of each entrypoint/function implemented in this interface"

Reply:
Addressed in https://github.com/pokt-network/pocket/pull/198/commits/46b1d648b52e445c09d228760df10363a8fff1c0